seth-shaw-unlv/controlled_access_terms
# Islandora tokens 1171

**GitHub Issue**: https://github.com/Islandora/documentation/issues/1171

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?
Consolidates the separate works from ASU and Jonathan Hunt to provide some islandora_tokens.

# What's new?
The code from https://github.com/kayakr/islandora_tokens and https://github.com/asulibraries/islandora-repo/tree/develop/web/modules/custom/islandora_tokens were merged into the the controlled_access_terms module.

Any tokens that were created with a previous install should still be supported by the new code (after legacy module is uninstalled and this is installed, for example, a token for [islandoratokens:agent_contributor] would still output the same value afterwards).

# How should this be tested?
After pulling in this branch and the cache is cleared, ANY place where you can use tokens should support these tokens, but we used it in the Metatag: Google Scholar in order to provide values for the various scholar meta tags. To use this, first install `metatag` and `metatag_google_scholar` and configure the "Repository item" content type admin/config/search/metatag/node__islandora_object to define several of the fields output to use some of the tokens. These should pop up when you click the "Browse available tokens." link -- and are all in the "Islandora Tokens" section.

# Interested parties
@Islandora/8-x-committers, @dannylamb, @kayakr
